tests/JsonReaderTests.cs

```diff
@@ -210,6 +210,52 @@ public void DateTime_With_Invalid_Input(string expectedErrorToken, string json)
                          expectedErrorToken);
     }
 
+    [Fact]
+    public void DateTimeOffset_Moves_Reader()
+    {
+        TestMovesReaderPastReadValue(JsonReader.DateTimeOffset(), "'2022-02-02T12:34:56-01:00'");
+    }
+
+    [Theory]
+    [InlineData(2022, 2, 2, 0, 0, 0, 0, null, null, "'2022-02-02'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, null, null, "'2022-02-02T12:34'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, 0, 0, "'2022-02-02T12:34Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, null, null, "'2022-02-02T12:34:56'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, 0, 0, "'2022-02-02T12:34:56Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, null, null, "'2022-02-02T12:34:56.078'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, 0, "'2022-02-02T12:34:56.078Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, 1, 0, "'2022-02-02T12:34:00+01:00'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, 2, 0, "'2022-02-02T12:34:56+02:00'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, -5, "'2022-02-02T12:34:56.078-00:05'")]
+    public void DateTimeOffset_With_Valid_Input(int year, int month, int day, int hour, int minute, int second, int millisecond, int? hourOffset, int? minuteOffset, string json)
+    {
+        var actual = JsonReader.DateTimeOffset().Read(Strictify(json));
+        var expectedOffset = hourOffset is { } h && minuteOffset is { } m ? new TimeSpan(h, m, 0) : DateTimeOffset.Now.Offset;
+        var expected = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, expectedOffset);
```

Comment:
I think using `DateTime.Now.Offset` could give the wrong results. For example, if the test data had used June for the month then the tests would have failed. In other words, the tests would start failing for a part of the year in time zones where daylight savings are observed. The following should fix this:

```suggestion
        var dateTime = new DateTime(year, month, day, hour, minute, second, millisecond);
        var expected = hourOffset is { } h && minuteOffset is { } m
                     ? new DateTimeOffset(dateTime, new TimeSpan(h, m, 0))
                     : dateTime;
```

The second part of the conditional expression will convert `DateTime` to `DateTimeOffset` and since `dateTime` is in local time, [`DateTimeOffset` will calculate the UTC offset based on the local time zone rules](https://github.com/dotnet/runtime/blob/f82fe1d83f1a0e22025e186fdc6f4d86de0a83a1/src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs#L88) (wherever the tests are running). In fact, I think we should expand the tests here for two months of the year (like June and December) that are generally known to vary in offsets in zones where daylight savings are observed.

Yet perhaps another way to simplify this whole test might be as follows:

```diff
diff --git a/tests/JsonReaderTests.cs b/tests/JsonReaderTests.cs
index 6a93ec1..aea258c 100644
--- a/tests/JsonReaderTests.cs
+++ b/tests/JsonReaderTests.cs
@@ -6,6 +6,7 @@ namespace Jacob.Tests;
 
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Linq;
 using System.Text;
 using System.Text.Json;
@@ -217,21 +218,30 @@ public void DateTimeOffset_Moves_Reader()
     }
 
     [Theory]
-    [InlineData(2022, 2, 2, 0, 0, 0, 0, null, null, "'2022-02-02'")]
-    [InlineData(2022, 2, 2, 12, 34, 0, 0, null, null, "'2022-02-02T12:34'")]
-    [InlineData(2022, 2, 2, 12, 34, 0, 0, 0, 0, "'2022-02-02T12:34Z'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 0, null, null, "'2022-02-02T12:34:56'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 0, 0, 0, "'2022-02-02T12:34:56Z'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 78, null, null, "'2022-02-02T12:34:56.078'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, 0, "'2022-02-02T12:34:56.078Z'")]
-    [InlineData(2022, 2, 2, 12, 34, 0, 0, 1, 0, "'2022-02-02T12:34:00+01:00'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 0, 2, 0, "'2022-02-02T12:34:56+02:00'")]
-    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, -5, "'2022-02-02T12:34:56.078-00:05'")]
-    public void DateTimeOffset_With_Valid_Input(int year, int month, int day, int hour, int minute, int second, int millisecond, int? hourOffset, int? minuteOffset, string json)
+    [InlineData("2022-02-02")]
+    [InlineData("2022-02-02T12:34")]
+    [InlineData("2022-02-02T12:34Z")]
+    [InlineData("2022-02-02T12:34:56")]
+    [InlineData("2022-02-02T12:34:56Z")]
+    [InlineData("2022-02-02T12:34:56.078")]
+    [InlineData("2022-02-02T12:34:56.078Z")]
+    [InlineData("2022-02-02T12:34:00+01:00")]
+    [InlineData("2022-02-02T12:34:56+02:00")]
+    [InlineData("2022-02-02T12:34:56.078-00:05")]
+    public void DateTimeOffset_With_Valid_Input(string input)
     {
-        var actual = JsonReader.DateTimeOffset().Read(Strictify(json));
-        var expectedOffset = hourOffset is { } h && minuteOffset is { } m ? new TimeSpan(h, m, 0) : DateTimeOffset.Now.Offset;
-        var expected = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, expectedOffset);
+        var formats = new[]
+        {
+            "yyyy-MM-dd",
+            "yyyy-MM-dd'T'HH:mm",
+            "yyyy-MM-dd'T'HH:mm:ss",
+            "yyyy-MM-dd'T'HH:mm:ss.fff",
+            "yyyy-MM-dd'T'HH:mmK",
+            "yyyy-MM-dd'T'HH:mm:ssK",
+            "yyyy-MM-dd'T'HH:mm:ss.fffK",
+        };
+        var expected = DateTimeOffset.ParseExact(input, formats, CultureInfo.InvariantCulture, DateTimeStyles.AssumeLocal);
+        var actual = JsonReader.DateTimeOffset().Read(Strictify($"'{input}'"));
         Assert.Equal(expected, actual);
     }
 
```